tools/lightning/overview-architecture.md

```diff
@@ -27,12 +27,14 @@ The complete import process is as follows:
 
 2. `tidb-lightning` creates the skeleton of all tables from the data source.
 
-3. For each table, `tidb-lightning` informs `tikv-importer` via gRPC to create an *engine file* to store KV pairs. `tidb-lightning` then reads the SQL dump in parallel, transforms the data into KV pairs according to the TiDB rules, and sends them to `tikv-importer`'s engine files.
+3. Each table is split into multiple continuous *batches*, so that data from a huge table (200 GB+) can be delivered incrementally.
 
-4. Once a full table of KV pairs are received, `tikv-importer` divides and schedules these data and imports them into the target TiKV cluster.
+4. For each batch, `tidb-lightning` informs `tikv-importer` via gRPC to create *engine files* to store KV pairs. `tidb-lightning` then reads the data source in parallel, transforms each row into KV pairs according to the TiDB rules, and sends them to `tikv-importer`'s engine files.
 
-5. `tidb-lightning` then performs a checksum comparison between the local data source and those calculated from the cluster, to ensure there is no data corruption in the process, and tells TiDB to `ANALYZE` all imported tables, to prepare for optimal query planning.
+5. Once a complete engine file is written, `tikv-importer` divides and schedules these data and imports them into the target TiKV cluster.
 
-6. After all tables are imported, `tidb-lightning` performs a global compaction on the TiKV cluster.
+    * There are two kinds of engine files: *data engines* and *index engines*, each corresponding to two kinds of KV pairs: the row data and secondary indices. Normally, the row data are entirely sorted in the data source, while the secondary indices are out of order. Because of this, the data engines are uploaded as soon as a batch is completed, while the index engines are imported only after all batches of the entire table are encoded.
```

Comment:
```suggestion
    There are two kinds of engine files: *data engines* and *index engines*, each corresponding to two kinds of KV pairs: the row data and secondary indices. Normally, the row data are entirely sorted in the data source, while the secondary indices are out of order. Because of this, the data engines are uploaded as soon as a batch is completed, while the index engines are imported only after all batches of the entire table are encoded.
```